### PR TITLE
New: Gson added to operate with Graph stored.

### PR DESCRIPTION
**Pull request by IlyaMoskva**
Common part to save/load graph to/from file.
Issue #41, which needed for Issues #42 and #43.

### CODE REVIEW

File: src/main/com/cqfn/degitx/simulator/data/GraphData.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    val tail: String
+)
+
+data class Node(
+    val hw: Hw,
+    val id: String,
+    val state: String
+)
+
+data class Hw(
```
Comment (stepanvalyavskiy) [inline src/main/com/cqfn/degitx/simulator/data/GraphData.kt:26]: Please rename it to Hardware, It takes a couple of seconds to understand what it is.

**Review by stepanvalyavskiy — CHANGES_REQUESTED** [review]
Hi, @IlyaMoskva, please check comments and add empty lines in the end of files.

File: src/main/com/cqfn/degitx/simulator/data/GraphData.kt (hunk 1 above)
Comment (IlyaMoskva) [reply src/main/com/cqfn/degitx/simulator/data/GraphData.kt]: Agree. Thank you, @stepanvalyavskiy , it is fixed.

**Review by stepanvalyavskiy — APPROVED** [review]
@IlyaMoskva, thanks

**Pull request merged**